.github/workflows/docs.yaml

```diff
@@ -5,30 +5,51 @@ on:
       - main
     tags:
       - 'v*'
+  pull_request:
 
 jobs:
   deploy:
+    name: Hugo
     runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}/go
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/rotationalio/ensign/docs
     steps:
       - name: Checkout Code
         uses: actions/checkout@v3
         with:
+          path: ${{ env.GOPATH }}/src/github.com/rotationalio/ensign
           submodules: true   # Fetch Hugo themes
           fetch-depth: 0     # Fetch all history for .GitInfo and .Lastmod
 
       - name: Setup Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.101.0'
+          hugo-version: '0.110.0'
           extended: true
 
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18.x.x"
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: Install NPM Dependencies
+        run: npm install
```

Comment:
I wonder if we need to specify npm version too? It looks like they just released a new version 4 days ago https://www.npmjs.com/package/npm?activeTab=versions